theme/admin/builder.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/*
+  # Common builder core
+
+  - adds settings page, but only if function mango_deploy_getDataset exists
+  - triggers deploy on a builder instance
+  - shows status
+  - provides mango_expandPaths fnc to simplify data structure (by __ delimiter)
+  - requires https://github.com/manGoweb/builder running instance
+
+  ## How-To
+
+  1. Implement a payload contruction in the theme/functions.php file:
+
+		function mango_deploy_getDataset() {
+			return [
+				'templates' => [
+					'src/templates/*.jade',
+				],
+				'data' => [
+		 			'site' => mango_expandPaths(get_option('site_settings')),
+				],
+			];
+		}
+
+	2. Fill in configuration paremeters:builder options
+
+	3. Test it and profit
+
+ */
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\BadResponseException;
+
+function mango_escape($str)
+{
+	return htmlspecialchars($str, ENT_COMPAT | ENT_HTML5);
+}
+
+function mango_pipeTransformer($key, $value, $transformers) {
+	if(!empty($transformers[$key])) {
+		return $transformers[$key]($value);
+	}
+	return $value;
+}
+
+function mango_expandPaths($root, $delimiter = NULL, $transformers = []) {
+	if(!$delimiter) {
+		$delimiter = '__';
+	}
+	if(!is_array($root)) {
+		return $root;
+	}
+	$updated = [];
+	foreach($root as $key => $val) {
+		$keyParts = explode($delimiter, (string) $key);
+		if(is_string($key) && count($keyParts) > 1) {
+			$firstKeyPart = array_shift($keyParts);
+			$keyRest = implode($delimiter, $keyParts);
+			$updated[$firstKeyPart] = isset($updated[$firstKeyPart]) ? $updated[$firstKeyPart] : [];
```

Comment:
@Onset já mám tendenci nepoužívat zbytečně nejnovější featury - co kdyby to někdo potřeboval rozjet na nižší verzi php :D ale chápu, že tenhle důvod asi neni validíní - takže jsem pro použití `??` :)